packages/twenty-front/src/modules/workflow/components/WorkflowEditActionFormServerlessFunction.tsx

```diff
@@ -41,32 +94,74 @@ export const WorkflowEditActionFormServerlessFunction = (
       headerTitle="Code - Serverless Function"
       headerType="Code"
     >
-      <Select
-        dropdownId="workflow-edit-action-function"
-        label="Function"
-        fullWidth
-        value={props.action.settings.input.serverlessFunctionId}
-        options={availableFunctions}
-        disabled={props.readonly}
-        onChange={(serverlessFunctionId) => {
-          if (props.readonly === true) {
-            return;
-          }
-
-          props.onActionUpdate({
-            ...props.action,
-            settings: {
-              ...props.action.settings,
-              input: {
-                serverlessFunctionId,
-                serverlessFunctionVersion:
-                  serverlessFunctions.find((f) => f.id === serverlessFunctionId)
-                    ?.latestVersion || 'latest',
-              },
-            },
-          });
-        }}
+      <Controller
+        name="serverlessFunctionId"
+        control={form.control}
+        render={({ field }) => (
+          <Select
+            dropdownId="select-serverless-function-id"
+            label="Function"
+            fullWidth
+            value={field.value}
+            options={availableFunctions}
+            disabled={props.readonly}
+            onChange={(serverlessFunctionId) => {
+              field.onChange(serverlessFunctionId);
+
+              const serverlessFunction = serverlessFunctions.find(
+                (f) => f.id === serverlessFunctionId,
+              );
+
+              const serverlessFunctionVersion =
+                serverlessFunction?.latestVersion || 'latest';
+
+              const defaultFunctionInput = serverlessFunction?.inputSchema
+                ? serverlessFunction.inputSchema
+                    .map((parameter) => parameter.name)
+                    .reduce((acc, name) => ({ ...acc, [name]: null }), {})
+                : {};
+
+              form.handleSubmit(async () => {
+                if (props.readonly === true) {
+                  return;
+                }
+
+                props.onActionUpdate({
+                  ...props.action,
+                  settings: {
+                    ...props.action.settings,
+                    input: {
+                      serverlessFunctionId,
+                      serverlessFunctionVersion,
+                      ...defaultFunctionInput,
+                    },
+                  },
+                });
+              })();
```

Comment:
Usually, we call `form.handleSubmit` once per form. What's the objective of calling it here? Is it to trigger validation?

Reply:
I remove react hook form. If we need validation, we will used zod directly. Not sure we have examples in the code though